### ISSUE TITLE
[java] java code generation compilation error

### ISSUE THREAD

**Issue by tamilmani** [swagger-api/swagger-codegen#2841]
Below is my swagger 2.0 code

```
  test:
    type: object
    properties:
      '@class':
        type: string
```

Generated enum class
     `public enum _classEnum { }`

Generated getter and setter methods
```
	@ApiModelProperty(example = "null", value = "Amount type")
	@JsonProperty("@class")
	public _classEnum getClass() {
		return _class;
	}
	public void setClass(_classEnum _class) {
		this._class = _class;
	}

```
getClass() method throws compilation error, because java consider to override the getClass() method.

Eclipse java compilation Error:
   The return type is incompatible with Object.getClass()

**Comment by wing328** [comment]
@tamilmani thanks for reporting the issue. To address it, we can map `class` to propertyClass so that the getter would become `getPropertyClass`.

**Comment by wing328** [comment]
Fixed via #3177. Please pull the latest master and give it a try.
